rocky/rocky/views/scheduler.py

```diff
@@ -61,12 +75,49 @@ def get_task_list(self) -> LazyTaskList | list[Any]:
             messages.error(self.request, error.message)
         return []
 
+    def get_report_schedule_form(self) -> ReportScheduleForm:
+        return self.report_schedule_form(self.request.POST)
+
+    def get_report_schedule_form_data(self):
+        form_data = self.get_report_schedule_form().data.dict()
+        return {k: v for k, v in form_data.items() if v}
+
     def get_task_details(self, task_id: str) -> Task | None:
         try:
             return self.scheduler_client.get_task_details(task_id)
         except SchedulerError as error:
             return messages.error(self.request, error.message)
 
+    def create_report_schedule(self, report_ooi, start_date: str, recurrence: str) -> ScheduleResponse | None:
+        try:
+            self.bytes_client.get_raw(report_ooi.data_raw_id)
```

Comment:
bytes won't be necessary anymore once we crate the recipe. The recipe will recreate the data and add it to bytes.